### ISSUE TITLE
Do not call requestPause() on AAudio if already paused

### ISSUE THREAD

**Issue by philburk** [google/oboe#65]
To avoid state transition problems in AAudio MMAP code. In MR1 this could leave you in a PAUSING state.

See spreadsheet for other transitions to avoid.
Make version dependent.

**Comment by dturner** [comment]
@philburk Do you have repro steps or a bug id for this? I [wrote a test](https://github.com/google/oboe/commit/71b6d41f2d82a1eccbbbccbde2d1f0d2c2242fa5) which I hoped would reproduce this bug and ran it on an emulator running Oreo MR1 (API 27) but it passed.

**Comment by dturner** [comment]
From Phil: This must be tested on an MMAP stream.

**Comment by philburk** [comment]
I found that I had already implemented a fix. It does not cover all state transitions.
For example requestPause() when Stopped will still transition to Pausing instead of Paused.

https://github.com/google/oboe/pull/191

REPRODUCE bug with NativeOboe Tester on Marlin on OC_MR1
click TEST OUTPUT
click OPEN
click START
click PAUSE
note that it displays "state = 6"
click PAUSE

EXPECT
state = 6

ACTUAL
state = 5
